### PR TITLE
update to accept parameters

### LINKED ISSUE #239: parameterize queries
the ability to parameterize queries would be extremely helpful, something like

```
?q=select foo from bar where bat=? and baz = ?&param[0]=$something$fiz$something$&param[1]=9
```

currently I've been able to do a somewhat hacky method of parameterizing via prepared statements but I got some feedback from you guys that that could lead to issues with pgbouncer

### PR DESCRIPTION
**Pull request by calvinmetcalf**
requires CartoDB/node-cartodb-psql#6 fixes #239

### CODE REVIEW

**Comment by calvinmetcalf** [comment]
will add tests

File: app/controllers/app.js [hunk 1]
```diff
@@ -233,12 +234,13 @@ function handleQuery(req, res) {
...
-        sql       = (sql      === "" || _.isUndefined(sql))      ? null : sql;
-        limit     = (!_.isNaN(limit))  ? limit : null;
-        offset    = (!_.isNaN(offset)) ? offset * limit : null;
+        dp           = (dp       === "" || _.isUndefined(dp))       ? '6'  : dp;
+        format       = (format   === "" || _.isUndefined(format))   ? 'json' : format.toLowerCase();
+        filename     = (filename === "" || _.isUndefined(filename)) ? 'cartodb-query' : sanitize_filename(filename);
+        sql          = (sql      === "" || _.isUndefined(sql))      ? null : sql;
+        limit        = (!_.isNaN(limit))  ? limit : null;
+        offset       = (!_.isNaN(offset)) ? offset * limit : null;
+        queryParams  = (Array.isArray()) ? queryParams : [];
```
Comment (rochoa) [inline app/controllers/app.js]: Missing `queryParams` in `Array.isArray`? :smile:

Comment (calvinmetcalf) [inline app/controllers/app.js]: thanks fixed

**Comment by Algunenano** [comment]
Stale

**Pull request closed without merging**